crates/ra_assists/src/raw_string.rs

```diff
@@ -0,0 +1,326 @@
+    let literal = ctx.node_at_offset::<Literal>()?;
+    if literal.token().kind() == ra_syntax::SyntaxKind::RAW_STRING {
+        ctx.add_action(AssistId("make_usual_string"), "make usual string", |edit| {
+            let text = literal.syntax().text();
+            let usual_start_pos = text.find_char('"').unwrap(); // we have a RAW_STRING
+            let end = literal.syntax().text_range().end();
+            dbg!(&end);
+            let mut i = 0;
+            let mut pos = 0;
+            let mut c = text.char_at(end - TextUnit::from(i));
+            while c != Some('"') {
+                if c != None {
+                    pos += 1;
+                }
+                i += 1;
+                c = text.char_at(end - TextUnit::from(i));
+            }
+
+            edit.target(literal.syntax().text_range());
+            edit.delete(TextRange::from_to(
+                literal.syntax().text_range().start(),
+                literal.syntax().text_range().start() + usual_start_pos,
+            ));
+            edit.delete(TextRange::from_to(
+                literal.syntax().text_range().end() - TextUnit::from(pos),
+                literal.syntax().text_range().end(),
+            ));
+        });
+        ctx.build()
+    } else {
+        None
+    }
+}
+
+pub(crate) fn add_hash(mut ctx: AssistCtx<impl HirDatabase>) -> Option<Assist> {
+    let literal = ctx.node_at_offset::<Literal>()?;
+    if literal.token().kind() == ra_syntax::SyntaxKind::RAW_STRING {
+        ctx.add_action(AssistId("add_hash"), "add hash to raw string", |edit| {
+            edit.target(literal.syntax().text_range());
+            edit.insert(literal.syntax().text_range().start() + TextUnit::from(1), "#");
+            edit.insert(literal.syntax().text_range().end(), "#");
+        });
+        ctx.build()
+    } else {
+        None
+    }
+}
+
+pub(crate) fn remove_hash(mut ctx: AssistCtx<impl HirDatabase>) -> Option<Assist> {
+    let literal = ctx.node_at_offset::<Literal>()?;
+    if literal.token().kind() == ra_syntax::SyntaxKind::RAW_STRING {
+        if !literal.syntax().text().contains_char('#') {
+            return None;
+        }
+        ctx.add_action(AssistId("remove_hash"), "remove hash from raw string", |edit| {
+            edit.target(literal.syntax().text_range());
+            edit.delete(TextRange::from_to(
+                literal.syntax().text_range().start() + TextUnit::from(1),
+                literal.syntax().text_range().start() + TextUnit::from(2),
+            ));
+            edit.delete(TextRange::from_to(
+                literal.syntax().text_range().end() - TextUnit::from(1),
+                literal.syntax().text_range().end(),
+            ));
+        });
+        ctx.build()
+    } else {
+        None
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::helpers::{check_assist, check_assist_not_applicable, check_assist_target};
+
+    #[test]
+    fn make_raw_string_target() {

```

Comment:
Could you please add some tests with for example `"` in them, or `\n`?

Reply:
A case is not currently managed: applying `remove_hash` on `r#"aaa"bbb"#` will make a broken string…  I'm trying to think of how to handle this